### DISCUSSION TITLE
Adding comments is a risky proposition

### DISCUSSION THREAD

**Discussion by marcusrbrown** [community/community#113676] [Mobile]
Hi,

- Don't dismiss or navigate away from an active comment editor when launching GitHub mobile, especially when launched via intent or a deep link (i.e., clicking on a GitHub link in another app).
- Absolutely do not dismiss an active comment editor from a swipe down gesture unless you offer a confirmation that the user intended to dismiss it.

I was writing a comment to share a link from another GitHub repo that I had opened through Safari. By default, clicking the repo link in Safari launched GitHub as expected, but using "Back" to navigate did not return me to my active comment editor.

In a separate comment editor, I swiped down to dismiss the onscreen keyboard, but the app dismissed it instead.

<details>
<summary>ℹ️ Additional Debug Info</summary>
GitHub for iOS, Version 1.154.0 (125806213)
Device: iPhone 14 Pro (iOS 17.4)
</details>

**Comment by esahmed21** [comment]
Thank you @marcusrbrown for brining this to our attention. I'll promptly create an internal ticket for our team to investigate improve the experience. Also sharing for awareness that your comment message will be saved as draft so when you get back to the place where you wrote the comment and opened the sheet, you should see a draft of your last message. This might help not losing your content, so let me know if this was not the case.

Thank you again for getting in touch!

**Reply by ipbrennan90** [reply]
This is absolutely still an issue on mobile. I was quoting a reply, wrote a long reply that took me a while to construct, upon swiping down it was destroyed. When going back to the comment and selecting "quote reply" the message was gone. I tested this again and the same thing happened. 

This is incredibly frustrating, a fix would be much appreciated or at the very least a warning that all the work will be destroyed when I do the swipe down gesture (this is unintentional 99% of the time)

**Reply by Alipalang137676** [reply]
Good
